packages/notifier/src/subscriber.js

```diff
@@ -51,51 +24,117 @@ const makeSubscriptionIterator = tailP => {
   });
 };
 
+/**
+ * Makes a behavioral presence of a possibly far subscription.
+ *
+ * @template T
+ * @param {GetUpdatesSince<T>} getUpdatesSince
+ * @returns {Subscription<T>}
+ */
+const makeSubscription = getUpdatesSince => {
+  return Far('Subscription', {
+    [Symbol.asyncIterator]: () => makeSubscriptionIterator(getUpdatesSince()),
+
+    getUpdatesSince,
+  });
+};
+
+/**
+ * Makes a behavioral presence of a possibly far subscription.
+ *
+ * @template T
+ * @param {ERef<Subscription<T>>} subscriptionP
+ * @returns {Subscription<T>}
+ */
+export const shadowSubscription = subscriptionP => {
+  const getUpdatesSince = (updateCount = NaN) =>
+    E(subscriptionP).getUpdatesSince(updateCount);
+
+  return makeSubscription(getUpdatesSince);
+};
+harden(shadowSubscription);
+
 /**
  * Makes a `{ publication, subscription }` for doing lossless efficient
  * distributed pub/sub.
  *
  * @template T
  * @returns {SubscriptionRecord<T>}
  */
-const makeSubscriptionKit = () => {
-  /** @type {((internals: ERef<SubscriptionInternals<T>>) => void) | undefined} */
-  let rear;
-  const hp = new HandledPromise(r => (rear = r));
-  const subscription = makeSubscription(hp);
+export const makeSubscriptionKit = () => {
+  /** @type {((internals: SubscriptionUpdates<T>) => void) | undefined} */
+  let tailR;
+  let tailP = new HandledPromise(r => (tailR = r));
+
+  let currentUpdateCount = 1;
+  /** @type {SubscriptionUpdates<T>} */
+  let currentP = tailP;
+  const advanceCurrent = () => {
+    // If tailP has not advanced past currentP, do nothing.
+    if (currentP !== tailP) {
+      currentUpdateCount += 1;
+      currentP = tailP;
+    }
```

Comment:
Style suggestion:
```suggestion
    if (currentP === tailP) {
      return;
    }
    currentUpdateCount += 1;
    currentP = tailP;
```

Reply:
Nice. This way of saying it also gives a more natural place to put the comment.